engine/runner/src/main/java/org/enso/runner/Main.java

```diff
@@ -1334,6 +1336,52 @@ void stderr(String msg) {
     System.err.println(msg);
   }
 
+  /**
+   * Checks if JVM mode should be enabled in a project defined by arguments, based on a project's
+   * config file, if any.
+   *
+   * @param line parsed command line arguments
+   * @return true, if project should be launched in JVM mode, false otherwise
+   */
+  private boolean isJvmModeEnabled(CommandLine line) {
+    var it = line.iterator();
+    String target = null;
+    while (target == null && it.hasNext()) {
+      var opt = it.next();
+      if (opt.getLongOpt().equals("run")) target = opt.getValue();
+    }
+    if (target != null) {
+      return jvmEnabledInProject(target);
+    } else {
+      return false;
+    }
+  }
+
+  private boolean jvmEnabledInProject(String runPath) {
+    var f = new File(runPath);
+    // Guess project's root directory
+    File configFile = null;
+    while (configFile == null && f != null) {
+      var testFile = f.toPath().resolve(org.enso.pkg.Config.ensoPackageConfigName());
+      if (testFile.toFile().exists()) {
+        configFile = testFile.toFile();
+      } else {
+        f = f.getParentFile();
+      }
+    }
+    if (configFile == null) {
+      return false;
+    } else {
+      try (FileReader fileReader = new FileReader(configFile)) {
```

Comment:
We are reading the config here and then the _running_ will read it again, right? Inefficiency (without functional impact).

Reply:
Yes, I'm well aware. Not the only place where I think we have such small inefficiency, I believe.